src/routes/amt/getHardwareInfo.ts

```diff
@@ -0,0 +1,145 @@
+/*********************************************************************
+* Copyright (c) Intel Corporation 2019
+* SPDX-License-Identifier: Apache-2.0
+* Description: Handler to get amt device audit logs
+**********************************************************************/
+
+import { Response, Request } from 'express'
+import { logger as log, logger } from '../../utils/logger'
+import { ErrorResponse } from '../../utils/amtHelper'
+import { MqttProvider } from '../../utils/MqttProvider'
+import { devices } from '../../server/mpsserver'
+
+export async function hardwareInfo (req: Request, res: Response): Promise<void> {
+  try {
+    const guid: string = req.params.guid
+
+    MqttProvider.publishEvent('request', ['AMT_HardwareInfo'], 'Hardware Information Requested', guid)
+    const response = await get(guid)
+    if (Object.keys(response).length === 0) {
+      log.error(`Request failed during AMTHardware Information BatchEnum Exec for guid : ${guid}.`)
+      MqttProvider.publishEvent('fail', ['AMT_HardwareInfo'], 'Failed to Get Hardware Information', guid)
+      return res.status(400).json(ErrorResponse(400, `Request failed during AMTHardware Information BatchEnum Exec for guid : ${guid}.`)).end()
+    } else {
+      MqttProvider.publishEvent('success', ['AMT_HardwareInfo'], 'Sent Hardware Information', guid)
+      logger.info(JSON.stringify(response, null, '\t'))
+      res.status(200).json(formatResponse(response)).end()
+    }
+  } catch (error) {
+    log.error(`Exception in AMT HardwareInformation : ${error}`)
+    MqttProvider.publishEvent('fail', ['AMT_HardwareInfo'], 'Interanl Server Error')
+    res.status(500).json(ErrorResponse(500, 'Request failed during AMTHardware Information.')).end()
+  }
+}
+
+export async function get (guid: string): Promise<any> {
+  const response: {[key: string]: any} = {}
+  const computerSystemPackage = await devices[guid].getComputerSystemPackage()
+  if (computerSystemPackage != null) {
+    response.CIM_ComputerSystemPackage = computerSystemPackage
+  }
+  const chassis = await devices[guid].getChassis()
+  if (chassis != null) {
```

Comment:
I'm not sure all these if conditions are necessary --- i think its fine to return the property set to null -- i would say thats better.

Reply:
done